shared/src/question-generators/math/modTricks.ts

```diff
@@ -0,0 +1,292 @@
+            : { correct: false, feedbackText: t(translations, lang, "feedbackIncorrect") };
+    };
+}
+
+// Exponentiation
+function generateExponentiationQuestion(lang: Language, path: string, random: Random) {
+    const a = random.int(2, 10);
+    const b = random.int(2, 10);
+    const n = random.int(2, 20);
+    const result = modularExponentiation(a, b, n);
+
+    const question: FreeTextQuestion = {
+        type: "FreeTextQuestion",
+        name: ModTricks.name(lang),
+        path: path,
+        text: t(translations, lang, "expQuestion", { a: String(a), b: String(b), n: String(n) }),
+        feedback: getExponentiationFeedbackFunction(lang, result),
+    };
+
+    const testing = { a, b, n, result };
+    return { question, testing };
+}
+
+function getExponentiationFeedbackFunction(lang: Language, correctAnswer: number): FreeTextFeedbackFunction {
+    return ({ text }) => {
+        const userAnswer = parseFloat(text.trim());
+        if (isNaN(userAnswer) || !Number.isInteger(userAnswer)) {
+            return { correct: false, feedbackText: t(translations, lang, "feedbackInvalid") };
+        }
+        return userAnswer === correctAnswer
+            ? { correct: true, feedbackText: t(translations, lang, "feedbackCorrect") }
+            : { correct: false, feedbackText: t(translations, lang, "feedbackIncorrect") };
+    };
+}
+
+// Chinese Remainder Theorem
+function generateCRTQuestion(lang: Language, path: string, random: Random) {
+    //determine number of congruences
+    const numCongruences = random.int(2, 4);
+    // congruence with a as remainder within (1, 20) and n as modulus within (2, 20)
+    const congruences: { a: number; n: number }[] = [];
+    const text: string[] = [];
+
+    // generate system of congruences
+    for (let i = 0; i < numCongruences; i++) {
+        const a = random.int(1, 20);
+        let n: number;
+
+        // find new modulus that is coprime to preceding moduli
+        do {
+            n = random.int(2, 20);
+        } while (congruences.some(({ n: precedingModulus }) => !areCoprime(n, precedingModulus)));
+
+        congruences.push({ a, n });
+        text.push(`x ≡ ${a} \\pmod{ ${n} }`);
+    }
+
+    const crtValue = solveCRT(congruences);
+    const commonModulus = congruences.reduce((acc, { n }) => acc * n, 1);
+
+    const question: FreeTextQuestion = {
+        type: "FreeTextQuestion",
+        name: ModTricks.name(lang),
+        path: path,
+        text: t(translations, lang, "crtQuestion", { text: text.join(", \\\\") }),
+        prompt: `$x ≡ $`,
+        bottomText: t(translations, lang, "crtBottomText"),
+        feedback: getCRTFeedbackFunction(lang, crtValue, commonModulus),
+    };
+
+    const testing = { crtValue, commonModulus };
+    return { question, testing };
+}
+
+function getCRTFeedbackFunction(lang: Language, crtValue: number, commonModulus: number): FreeTextFeedbackFunction {
```

Comment:
Maybe also a check format function? So the user can see if the input has the correct type